mlflow/exceptions.py

```diff
@@ -169,17 +170,32 @@ def __init__(self):
         super().__init__(self.MESSAGE, error_code=NOT_IMPLEMENTED)
 
 
-class MlflowTraceDataNotFound(MlflowException):
+class MlflowTraceDataException(MlflowException):
+    """Exception thrown for trace data related error"""
+
+    def __init__(
+        self, error_code: str, request_id: Optional[str] = None, artifact_path: Optional[str] = None
+    ):
+        if request_id:
+            self.msg = f"request_id={request_id}"
+        elif artifact_path:
+            self.msg = f"path={artifact_path}"
+
+        if error_code == NOT_FOUND:
+            super().__init__(f"Trace data not found for {self.msg}", error_code=error_code)
+        elif error_code == INVALID_STATE:
+            super().__init__(f"Trace data is corrupted for {self.msg}", error_code=error_code)
```

Comment:
/ai `msg` doesn't sound right to store request_id or path. Suggest a few different names.

Reply:
/ai Thanks, but the suggested names are all too long.

Reply:
@serena-ruan can you use `id`?

Reply:
I choose ctx, is that ok?